### PR TITLE
vendir: init at 0.17.0

### PR DESCRIPTION
**Pull request by russell**
###### Motivation for this change

Vendir is a tool for vendoring files into a git repository, it works well with tooling that otherwise doesn't provide mechanisms to import files from other repositories like Jsonnet.

Vendir is part of the carvel collection of kubernetes tooling with ytt and kapp which are already in nixpkgs 

###### Things done

- [x] Tested using sandboxing ([nix.useSandbox](https://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](https://nixos.org/nix/manual/#sec-conf-file) on non-NixOS linux)
- Built on platform(s)
   - [x] NixOS
   - [ ] macOS
   - [x] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [ ] Tested compilation of all pkgs that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review wip"`
- [x] Tested execution of all binary files (usually in `./result/bin/`)
- [ ] Determined the impact on package closure size (by running `nix path-info -S` before and after)
- [ ] Ensured that relevant documentation is up to date
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

### CODE REVIEW

**Comment by russell** [comment]
I didn't manage to test via a sandbox, I built the package, then couldn't figure out how to rebuild it, it would always use the version in the cache.

**Comment by r-rmcgibbo** [comment]
Result of `nixpkgs-review pr 117824` at f1256268 run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>1 package built successfully:</summary>
  <ul>
    <li>vendir</li>
  </ul>
</details>
<details>
  <summary>1 suggestion:</summary>
  <ul>
    <li>warning: unused-argument

Unused argument: `installShellFiles`.
Near pkgs/development/tools/vendir/default.nix:1:40:
```
  |
1 | { lib, buildGoModule, fetchFromGitHub, installShellFiles }:
  |                                        ^
```
</li>
  </ul>
</details>

File: maintainers/maintainer-list.nix [hunk 1]
```diff
@@ -8327,6 +8327,12 @@
     githubId = 4958190;
     name = "Rushmore Mushambi";
   };
+  russell = {
```
Comment (SuperSandro2000) [inline maintainers/maintainer-list.nix:8330]: Please put this into a separate commit with the message ``maintainers: add russell``.

File: pkgs/development/tools/vendir/default.nix [hunk 2]
```diff
@@ -0,0 +1,23 @@
+{ lib, buildGoModule, fetchFromGitHub }:
+buildGoModule rec {
```
Comment (SuperSandro2000) [inline pkgs/development/tools/vendir/default.nix:2]:
```suggestion
{ lib, buildGoModule, fetchFromGitHub }:

buildGoModule rec {
```

Comment (russell) [reply pkgs/development/tools/vendir/default.nix]: thanks, applied

File: maintainers/maintainer-list.nix (hunk 1 above)
Comment (russell) [reply maintainers/maintainer-list.nix:8330]: no problem, split into 2 commits :)

**Comment by SuperSandro2000** [comment]
> I didn't manage to test via a sandbox, I built the package, then couldn't figure out how to rebuild it, it would always use the version in the cache.

``nix-build -A vendor --check`` rebuilds it.

**Comment by russell** [comment]
> > I didn't manage to test via a sandbox, I built the package, then couldn't figure out how to rebuild it, it would always use the version in the cache.
> 
> `nix-build -A vendor --check` rebuilds it.

Worked like a charm, thank you
